stout/http.h

```diff
@@ -0,0 +1,524 @@
+      Eventual<Result>()
+          .context(Data(loop, url, method, timeout, fields_string))
+          .start([](auto& data, auto& k) {
+            using K = std::decay_t<decltype(k)>;
+            data.k = &k;
+
+            uv_handle_set_data(data.timer_handle.base_handle(), &data);
+
+            static auto check_multi_info = [](Data& data) {
+              CURLMsg* message = nullptr;
+              int msgq = 0;
+
+              long response = 0;
+              auto& k = *static_cast<K*>(data.k);
+
+              while ((message = curl_multi_info_read( //
+                          data.multi.handle(),
+                          &msgq))) {
+                switch (message->msg) {
+                  case CURLMSG_DONE:
+                    if (message->data.result == CURLE_OK) {
+                      curl_easy_getinfo(
+                          data.easy.handle(),
+                          CURLINFO_RESPONSE_CODE,
+                          &response);
+                      uv_timer_stop(data.timer_handle.handle());
+                      data.curl_context_map.clear();
+                      k.Start(Result{
+                          data.context.buffer.Extract(),
+                          response});
+                      break;
+                    } else {
+                      uv_timer_stop(data.timer_handle.handle());
+                      data.curl_context_map.clear();
+                      k.Fail(curl_easy_strerror(message->data.result));
+                      break;
+                    }
+
+                    CHECK_EQ(
+                        curl_multi_remove_handle(
+                            data.multi.handle(),
+                            message->easy_handle),
+                        CURLM_OK);
+                    break;
+
+                  default:
+                    CHECK_EQ(
+                        curl_multi_remove_handle(
+                            data.multi.handle(),
+                            message->easy_handle),
+                        CURLM_OK);
+                    uv_timer_stop(data.timer_handle.handle());
+                    data.curl_context_map.clear();
+                    k.Fail(curl_easy_strerror(CURLE_ABORTED_BY_CALLBACK));
+                    break;
+                }
+              }
+            };
+
+            static auto poll_callback = [](
+                                            uv_poll_t* handle,
+                                            int status,
+                                            int events) {
+              Data& data = *static_cast<Data*>(handle->data);
+
+              uv_timer_stop(data.timer_handle.handle());
+              int running_handles;
+              int flags = 0;
+
+              if (status < 0)
+                flags = CURL_CSELECT_ERR;
+              if (!status && (events & UV_READABLE))
+                flags |= CURL_CSELECT_IN;
+              if (!status && (events & UV_WRITABLE))
+                flags |= CURL_CSELECT_OUT;
```

Comment:
Please use curly braces for all if-statements.

Reply:
Fixed!